nixos/modules/services/security/soarca.nix

```
@@ -0,0 +1,117 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.services.soarca;
+in
+{
+  options.services.soarca = {
+    enable = lib.mkEnableOption "SOARCA";
+    package = lib.mkPackageOption pkgs "soarca" { };
+
+    settings = lib.mkOption {
+      type = lib.types.submodule {
+        freeformType =
+          with lib.types;
+          attrsOf (
+            nullOr (oneOf [
+              bool
+              int
+              str
+            ])
+          );
+        options = { };
```

I think it would be nice to spec some options out instead of only providing examples

```suggestion
        options = {
          PORT = lib.mkOption {
            type = lib.types.int;
            description = "Set the exposed port of SOARCA.";
            default = 8080;
            example = lib.literalExpression "9000";
          };
          
          GIN_MODE = lib.mkOption {
            type = lib.types.str;
            description = "Set the GIN mode.";
            default = "release";
          };
          
          DATABASE = lib.mkOption {
            type = lib.types.bool;
            description = "Set if you want to run with an external database.";
            default = false;
          };
        };
```